src/main/java/frc/robot/hardware/phoenix6/Phoenix6Device.java

```diff
@@ -31,17 +31,30 @@ public boolean isConnected() {
 		return connectedInput.connected;
 	}
 
+	private InputSignal<?>[] getValidSignals(InputSignal<?>... signals) {
+		LinkedList<InputSignal<?>> signalsSet = new LinkedList<>();
+		for (InputSignal<?> signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter) {
+				signalsSet.add(signal);
+			} else {
+				new Alert(
+					Alert.AlertType.WARNING,
+					logPath + "signal named " + signal.getName() + " got invalid type " + signal.getClass().getSimpleName()
+				).report();
+			}
```

Comment:
I Don't like the level of nesting here so consider moving into a seperate private function called `validateSignal` or something.

Reply:
its just creating alert, its long but like the name is long there is no reason for func